pandas/core/generic.py

```diff
@@ -1247,9 +1247,12 @@ def equals(self, other):
 
         This function allows two Series or DataFrames to be compared against
         each other to see if they have the same shape and elements. NaNs in
-        the same location are considered equal. The column headers do not
-        need to have the same type, but the elements within the columns must
-        be the same dtype.
+        the same location are considered equal.
+        
+        The row/column index do not need to have the same type, as long
+        as the values are still considered equal.
```

Comment:
I feel the "still" can be removed. This was used in a different context under "Notes". It makes the sentence slightly confusing when used here. So we will have "as long as the values are considered equal".

Reply:
@fujiaxiang added your suggestion, thanks!